libhdf5/hdf5var.c

```diff
@@ -886,6 +905,102 @@ NC4_def_var_deflate(int ncid, int varid, int shuffle, int deflate,
                             &deflate_level, NULL, NULL, NULL, NULL, NULL, NULL);
 }
 
+/**
+ * Set szip compression settings on a variable. Szip is an
+ * implementation of the extended-Rice lossless compression
+ * algorithm. Szip is reported to provide fast and effective
+ * compression.
+ *
+ * SZIP compression cannot be applied to variables with any
+ * user-defined type.
+ *
+ * @note The options_mask parameter may be either NC_SZIP_EC (entropy
+ * coding) or NC_SZIP_NN (nearest neighbor):
+ * * The entropy coding method is best suited for data that has been
+ * processed. The EC method works best for small numbers.
+ * * The nearest neighbor coding method preprocesses the data then the
+ * applies EC method as above.
+ *
+ * @param ncid File ID.
+ * @param varid Variable ID.
+ * @param options_mask The options mask. Can be NC_SZIP_EC or
+ * NC_SZIP_NN.
+ * @param pixels_per_block Pixels per block. Must be even and not
+ * greater than 32, with typical values being 8, 10, 16, or 32. This
+ * parameter affects compression ratio; the more pixel values vary,
+ * the smaller this number should be to achieve better performance. If
+ * pixels_per_block is bigger than the total number of elements in a
+ * dataset chunk, H5Pset_szip will succeed but the subsequent call to
+ * H5Dcreate will fail; the conflict can be detected only when the
+ * property list is used.
+ *
+ * @returns ::NC_NOERR No error.
+ * @returns ::NC_ENOTBUILT This HDF5 install was not built with szip.
+ * @returns ::NC_EBADID Bad ncid.
+ * @returns ::NC_ENOTVAR Invalid variable ID.
+ * @returns ::NC_ENOTNC4 Attempting netcdf-4 operation on file that is
+ * not netCDF-4/HDF5.
+ * @returns ::NC_ELATEDEF Too late to change settings for this variable.
+ * @returns ::NC_ENOTINDEFINE Not in define mode.
+ * @returns ::NC_EINVAL Invalid input
+ * @author Ed Hartnett
+ */
+int
+nc_def_var_szip(int ncid, int varid, int options_mask, int pixels_per_block)
+{
+    NC_GRP_INFO_T *grp;
+    NC_FILE_INFO_T *h5;
+    NC_VAR_INFO_T *var;
+    int built = 0;
+    int ret;
+
+    LOG((2, "%s: ncid 0x%x varid %d", __func__, ncid, varid));
+
+    /* If HDF5 was not built with szip, then return error. */
+#ifdef HAVE_H5Z_SZIP
```

Comment:
Where is this value defined?  Is it in an hdf5 include file or something set by config/cmake?

Reply:
It is defined in config.h.

I can also add something in netcdf_meta.h.

Reply:
It is set my cmake/configure. It's an already-existing macro that was used to handle nc_inq_var_szip(). It gets defined in config.h. It is also already in netcdf_meta.h as:
`#define NC_HAS_SZIP      1 /*!< szip support (HDF5 only) */`

It is not currently in netcdf.settings. I added it.

Reply:
Thanks.  Overlooked the `NC_HAS_SZIP` which should do what I need.